res/controllers/Traktor-Kontrol-S2-MK2-hid-scripts.js

```diff
@@ -0,0 +1,1336 @@
+  OutputBottom.addOutput("[Channel2]", "!pad_4_B", 0x18, "B");
+
+  this.controller.registerOutputPacket(OutputBottom);
+
+  // Link up control objects to their outputs
+  TraktorS2MK2.linkDeckOutputs("sync_enabled", TraktorS2MK2.outputCallback);
+  TraktorS2MK2.linkDeckOutputs("cue_indicator", TraktorS2MK2.outputCallback);
+  TraktorS2MK2.linkDeckOutputs("play_indicator", TraktorS2MK2.outputCallback);
+  
+  TraktorS2MK2.setPadMode("[Channel1]", TraktorS2MK2.pad_modes.hotcue);
+  TraktorS2MK2.setPadMode("[Channel2]", TraktorS2MK2.pad_modes.hotcue);
+  
+  TraktorS2MK2.linkDeckOutputs("loop_in", TraktorS2MK2.outputCallbackLoop);
+  TraktorS2MK2.linkDeckOutputs("loop_out", TraktorS2MK2.outputCallbackLoop);
+  TraktorS2MK2.linkDeckOutputs("keylock", TraktorS2MK2.outputCallbackDark);
+  TraktorS2MK2.linkDeckOutputs("LoadSelectedTrack", TraktorS2MK2.outputCallback);
+  TraktorS2MK2.linkDeckOutputs("slip_enabled", TraktorS2MK2.outputCallback);
+  TraktorS2MK2.linkChannelOutput("[Channel1]", "pfl", TraktorS2MK2.outputChannelCallback);
+  TraktorS2MK2.linkChannelOutput("[Channel2]", "pfl", TraktorS2MK2.outputChannelCallback);
+  TraktorS2MK2.linkChannelOutput("[Channel1]", "track_loaded", TraktorS2MK2.outputChannelCallback);
+  TraktorS2MK2.linkChannelOutput("[Channel2]", "track_loaded", TraktorS2MK2.outputChannelCallback);
+  TraktorS2MK2.linkChannelOutput("[Channel1]", "PeakIndicator", TraktorS2MK2.outputChannelCallback);
+  TraktorS2MK2.linkChannelOutput("[Channel2]", "PeakIndicator", TraktorS2MK2.outputChannelCallback);
+  TraktorS2MK2.linkChannelOutput("[EffectRack1_EffectUnit1]", "group_[Channel1]_enable", TraktorS2MK2.outputChannelCallback);
+  TraktorS2MK2.linkChannelOutput("[EffectRack1_EffectUnit2]", "group_[Channel1]_enable", TraktorS2MK2.outputChannelCallback);
+  TraktorS2MK2.linkChannelOutput("[EffectRack1_EffectUnit1]", "group_[Channel2]_enable", TraktorS2MK2.outputChannelCallback);
+  TraktorS2MK2.linkChannelOutput("[EffectRack1_EffectUnit2]", "group_[Channel2]_enable", TraktorS2MK2.outputChannelCallback);
+
+  engine.makeConnection("[EffectRack1_EffectUnit1]", "focused_effect", TraktorS2MK2.onFocusedEffectChange);
+  engine.makeConnection("[EffectRack1_EffectUnit2]", "focused_effect", TraktorS2MK2.onFocusedEffectChange);  
+  TraktorS2MK2.connectEffectButtonLEDs("[EffectRack1_EffectUnit1]");
+  TraktorS2MK2.connectEffectButtonLEDs("[EffectRack1_EffectUnit2]");
+  
+  engine.makeConnection("[Channel1]", "VuMeter", TraktorS2MK2.onVuMeterChanged);
+  engine.makeConnection("[Channel2]", "VuMeter", TraktorS2MK2.onVuMeterChanged);
+
+  engine.makeConnection("[Channel1]", "loop_enabled", TraktorS2MK2.onLoopEnabledChanged);
+  engine.makeConnection("[Channel2]", "loop_enabled", TraktorS2MK2.onLoopEnabledChanged);
+}
+
+TraktorS2MK2.linkDeckOutputs = function(key, callback) {
+  // Linking outputs is a little tricky because the library doesn't quite do what I want.  But this
+  // method works.
```

Comment:
these links look fine, is the comment out of date?

Reply:
I think you wrote that comment for the S4 Mk2 mapping :P

Reply:
Yeah maybe the code has been updated since then?  :P